common/src/main/java/com/wynntils/features/user/inventory/UnidentifiedItemIconFeature.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features.user.inventory;
+
+import com.mojang.blaze3d.vertex.PoseStack;
+import com.wynntils.core.features.UserFeature;
+import com.wynntils.core.features.properties.FeatureInfo;
+import com.wynntils.mc.event.HotbarSlotRenderEvent;
+import com.wynntils.mc.event.SlotRenderEvent;
+import com.wynntils.mc.render.RenderUtils;
+import com.wynntils.mc.render.Texture;
+import com.wynntils.wc.custom.item.UnidentifiedItemStack;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+@FeatureInfo(category = "Inventory")
+public class UnidentifiedItemIconFeature extends UserFeature {
+    @SubscribeEvent
+    public void onSlotRender(SlotRenderEvent.Post e) {
+        drawIcon(e.getSlot().getItem(), e.getSlot().x, e.getSlot().y);
+    }
+
+    @SubscribeEvent
+    public void onHotbarSlotRender(HotbarSlotRenderEvent.Post e) {
+        drawIcon(e.getStack(), e.getX(), e.getY());
+    }
+
+    private void drawIcon(ItemStack item, int slotX, int slotY) {
+        if (!(item instanceof UnidentifiedItemStack unidentifiedItem)) return;
+
+        RenderUtils.drawTexturedRect(
+                new PoseStack(),
+                Texture.GEAR_ICONS.resource(),
+                slotX + 2,
+                slotY + 2,
+                400,
+                12,
+                12,
+                unidentifiedItem.getItemType().getIconTextureX(),
+                unidentifiedItem.getItemType().getIconTextureY(),
+                16,
+                16,
+                Texture.GEAR_ICONS.width(),
+                Texture.GEAR_ICONS.height());
+    }
+}
```

Comment:
Maybe this feature should only contain configs, and the actual rendering etc should be done by ItemProperties?

Reply:
This is an analogous setup to the durability arc, item highlights and text overlay. 🤔
If we want a more generic approach for rendering texture on top of items we can,  but what else will it be used for?

Reply:
I am not sure, perhaps the others will have stronger opinions. I am fine with this being a feature or a item property impl.

Reply:
an ItemProperty still requires a feature to do rendering code - i think this implementation is fine; imo it's safe to assume that we won't need this functionality for any other type of item, and on the off chance that we do, we can make the necessary modifications